test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj

```diff
@@ -4,8 +4,4 @@
     <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
   </PropertyGroup>
 
-  <ItemGroup>
-    <Compile Include="..\..\..\IntegrationTests\Helpers\TcpPortProvider.cs" />
```

Comment:
Do we still need this file? It is kind of low risky but it seems better to allocate and then inform like you did in this PR.

Reply:
In general, it is used in all integration tests, here it was included to not copy+paste it to the test app. But the solution I propose in this PR should be more stable.

Reply:
yep, ideally we get to the point of deleting that source file.

Reply:
We would need to use something different than `HttpListener` to do that 😉

Reply:
@RassK Is close to achieve it https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1372